In [1]:
import sys
sys.path.append("..")
from pathlib import Path
import plotly.graph_objects as px
from utils.transcription_utils import get_transcription_info,get_stats,get_total_stats
import csv
import pandas as pd

In [2]:
save_path=Path("/media/dataset-harddisk/munikumar/utils/audio_utils/data/mucs/normalize")

In [3]:
dataset_dir=Path("/media/dataset-harddisk/munikumar/hindi_dataset/mucs/mucs_raw")

In [4]:
def load_transcription(transcription_path):

    with open(transcription_path, "r", encoding="utf-8") as fp:
        data = fp.read().splitlines()
        transcription_data = {
            transcription.split(" ")[0] + ".wav": " ".join(transcription.split(" ")[1:])
            for transcription in data
        }
    
    return transcription_data

In [5]:
test_transcription_data=load_transcription(dataset_dir/"subtask1_blindtest_wReadme"/"Hindi"/"transcription.txt")
get_transcription_info(test_transcription_data,dataset_dir/"subtask1_blindtest_wReadme"/"Hindi"/"audio",save_path,"test")
get_stats(save_path/"test_non_hindi_transcripts.json")


getting audio info about : normalize - test


  0%|          | 0/3897 [00:00<?, ?it/s]

100%|██████████| 3897/3897 [04:13<00:00, 15.40it/s]

**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:29:26
Loss duration : 0:0:0
Remaining duration : 5:29:26
Drop by percentage: 0.00

Total audio files : 3897
Files that would be drop: 0
Remaining audio files : 3897
Drop by percentage: 0.00
**************************************************
Non hindi tokes : []


### Test Dataset
**************************************************
**Loss of audio data without transcription normalization:**
**************************************************
Total duration : 5:29:26  
Loss duration : 1:54:17  
Remaining duration : 3:35:9  
Drop by percentage: 34.69  

Total audio files : 3897  
Files that would be drop: 1346  
Remaining audio files : 2551  
Drop by percentage: 34.54  
**************************************************
Non hindi tokes : ['ग़', 'ज़', 'ड़', 'ढ़', 'फ़']

**************************************************
**Loss of audio data After decomposing nuktas:**
**************************************************
Total duration : 5:29:26  
Loss duration : 0:0:0  
Remaining duration : 5:29:26  
Drop by percentage: 0.00  

Total audio files : 3897  
Files that would be drop: 0  
Remaining audio files : 3897  
Drop by percentage: 0.00  
**************************************************
Non hindi tokes : []

In [6]:
dev_transcription_data=load_transcription(dataset_dir/"test"/"transcription.txt")
get_transcription_info(dev_transcription_data,dataset_dir/"test"/"audio",save_path,"dev")
get_stats(save_path/"dev_non_hindi_transcripts.json")

getting audio info about : normalize - dev


  0%|          | 0/3843 [00:00<?, ?it/s]

100%|██████████| 3843/3843 [04:18<00:00, 14.89it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:33:25
Loss duration : 0:0:48
Remaining duration : 5:32:37
Drop by percentage: 0.24

Total audio files : 3843
Files that would be drop: 10
Remaining audio files : 3833
Drop by percentage: 0.26
**************************************************
Non hindi tokes : ['ऍ']


### Dev Dataset
**************************************************
**Loss of audio data without transcription normalization:**
**************************************************
Total duration : 5:33:25  
Loss duration : 1:11:37  
Remaining duration : 4:21:47  
Drop by percentage: 21.48  

Total audio files : 3843  
Files that would be drop: 781  
Remaining audio files : 3062  
Drop by percentage: 20.32  
**************************************************
Non hindi tokes : ['ऍ', 'क़', 'ख़', 'ज़', 'ड़', 'ढ़', 'फ़']

**************************************************
**Loss of audio data After Decomposing Nuktas:**
**************************************************
Total duration : 5:33:25  
Loss duration : 0:0:48  
Remaining duration : 5:32:37  
Drop by percentage: 0.24  

Total audio files : 3843  
Files that would be drop: 10  
Remaining audio files : 3833  
Drop by percentage: 0.26  
**************************************************
Non hindi tokes : ['ऍ']

In [7]:
train_transcription_data=load_transcription(dataset_dir/"train"/"transcription.txt")
get_transcription_info(train_transcription_data,dataset_dir/"train"/"audio",save_path,"train")
get_stats(save_path/"train_non_hindi_transcripts.json")

getting audio info about : normalize - train


 26%|██▌       | 25536/99925 [33:03<1:16:48, 16.14it/s]

### Train Dataset:
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 95:3:0  
Loss duration : 11:36:50  
Remaining duration : 83:26:9  
Drop by percentage: 12.22  

Total audio files : 99925  
Files that would be drop: 10851  
Remaining audio files : 89074  
Drop by percentage: 10.86  
**************************************************
Non hindi tokes : ['ऍ', 'क़', 'ख़', 'ग़', 'ज़', 'ड़', 'ढ़', 'फ़', 'ॠ']

**************************************************
Loss of audio data After Decomposing Nuktas:
**************************************************
Total duration : 95:3:0  
Loss duration : 0:18:26  
Remaining duration : 94:44:33  
Drop by percentage: 0.32  

Total audio files : 99925  
Files that would be drop: 268  
Remaining audio files : 99657  
Drop by percentage: 0.27  
**************************************************
Non hindi tokes : ['ऍ', 'ॠ']

In [ ]:
get_total_stats(save_path)

**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 106:5:51
Loss duration : 0:19:14
Remaining duration : 105:46:37
Drop by percentage: 0.30

Total audio files : 107665
Files that would be drop: 278
Remaining audio files : 107387
Drop by percentage: 0.26


### Total Dataset
**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 106:5:51  
Loss duration : 14:42:45  
Remaining duration : 91:23:6  
Drop by percentage: 13.87  

Total audio files : 107665  
Files that would be drop: 12978  
Remaining audio files : 94687  
Drop by percentage: 12.05  

**************************************************
Loss of audio data for total dataset After decomposing Nuktas:
**************************************************
Total duration : 106:5:51  
Loss duration : 0:19:14  
Remaining duration : 105:46:37  
Drop by percentage: 0.30  

Total audio files : 107665  
Files that would be drop: 278  
Remaining audio files : 107387  
Drop by percentage: 0.26  

## Raw audio duration

train_raw audio duration : 95:03:00  
dev_raw audio duration: 5:33:25  
test_raw audio duration: 5:29:26  
total_duration: 106:05:51

## Total number of audio files

train_audio_files: 99925  
dev_audio_files: 3843  
test_audio_files: 3897  
total_audio files: 107665
